hyperspy/_signals/complex_signal.py

```diff
@@ -262,15 +262,12 @@ def angle(self, deg=False):
         return super().angle(angle, deg=deg)
     angle.__doc__ = ComplexSignal_mixin.angle.__doc__
 
-    def argand_diagram(self, plot=True, size=[256, 256], display_range=None):
+    def argand_diagram(self, size=[256, 256], display_range=None):
```

Comment:
From a quick look, it seems that this function works fine for 1D and 2D signal but it does not take into account the navigation dimension.

Reply:
@thomasaarholt  It does work for signals with any dimensionality as it is just plotting of  data on Im(Re) axis.
@ericpre This is also correct it creates a diagram for the whole dataset at once rather than for slices in navigation dimensions.

Reply:
Ok, I would have expected some array reshaping but it may not be necessary, if everything is ordered properly when creating the `Signal2D` of the _argand_ signal.
If this is the case, it would be good to add a test checking the navigation axes (and their attributes) are correct, etc.
Complex signal can take a bit of space into memory, is it worth adding lazy processing?

Reply:
@ericpre Perhaps it would make sense adding possibility to apply the function to the whole dataset as well as to iterate it along navigation dimensions, what would you say?

Indeed, Lazy is missing. Will add it asap.

Reply:
I think this is fine as it is (providing that the tests with navigation dimension are OK). If this is not necessary, I don't see why we add the option to iterate on navigation axis.

Reply:
Sorry for the delay. It isn't straight forward to implement it for `dask` arrays. I'm still looking at how it can be done, but not very certain if I will manage

Reply:
No worries, it was a suggestion, we can merge it without lazy support if you prefer.
Indeed, there is no histogram2d, nor histogramdd in dask, but this https://github.com/dask/dask/issues/2911#issuecomment-422401799 may be useful.

Reply:
@ericpre Lazy now would rise `NotImplementedError`, and the test would checking it